js/apps/admin-ui/src/components/users/UserSelect.tsx

```diff
@@ -53,46 +52,66 @@ export const UserSelect = ({
   const values: string[] | undefined = getValues(name!);
 
   const [open, toggleOpen, setOpen] = useToggle();
-  const [users, setUsers] = useState<(UserRepresentation | undefined)[]>([]);
+  const [selectedUsers, setSelectedUsers] = useState<UserRepresentation[]>([]);
+  const [searchedUsers, setSearchedUsers] = useState<UserRepresentation[]>([]);
   const [inputValue, setInputValue] = useState("");
   const [search, setSearch] = useState("");
   const textInputRef = useRef<HTMLInputElement>();
 
   const debounceFn = useCallback(debounce(setSearch, 1000), []);
 
   useFetch(
-    () => {
-      const params: UserQuery = {
-        max: 20,
-      };
+    async () => {
+      if (!values) {
+        return [];
+      }
+
+      const foundUsers = await Promise.all(
+        values.map((id) => adminClient.users.findOne({ id })),
+      );
+
+      return foundUsers.filter((user) => user !== undefined);
+    },
+    setSelectedUsers,
+    [values],
+  );
 
-      if (search) {
-        params.username = search;
+  useFetch(
+    async () => {
+      if (!search) {
+        return [];
```

Comment:
If your not searching shouldn't it show the first couple of users? Seems a bit odd when you only click it to see an empty box:
![image](https://github.com/user-attachments/assets/edb3c88c-854e-45ed-9f8a-4bdb07c744d5)